pkg/security/resolvers/tags/resolver_linux.go

```diff
@@ -117,30 +131,107 @@ func (t *LinuxResolver) checkTags(pendingWorkload *Workload) {
 	}
 }
 
+// getWorkloadID returns the workload ID for a workload
+func (t *LinuxResolver) getWorkloadID(workload *Workload) interface{} {
+	if len(workload.ContainerID) != 0 {
+		return workload.ContainerID
+	}
+	return workload.CGroupID
+}
+
 // fetchTags fetches tags for the provided workload
 func (t *LinuxResolver) fetchTags(workload *Workload) error {
-	newTags, err := t.ResolveWithErr(workload.ContainerID)
+	workloadID := t.getWorkloadID(workload)
+	newTags, err := t.ResolveWithErr(workloadID)
 	if err != nil {
-		return fmt.Errorf("failed to resolve %s: %w", workload.ContainerID, err)
+		return fmt.Errorf("failed to resolve %v: %w", workloadID, err)
 	}
 
-	workload.Selector.Image = utils.GetTagValue("image_name", newTags)
-	workload.Selector.Tag = utils.GetTagValue("image_tag", newTags)
-	if len(workload.Selector.Image) != 0 && len(workload.Selector.Tag) == 0 {
-		workload.Selector.Tag = "latest"
+	workload.Tags = newTags
+
+	// For container workloads, try to extract image information
+	if len(workload.ContainerID) != 0 {
+		workload.Selector.Image = utils.GetTagValue("image_name", newTags)
+		workload.Selector.Tag = utils.GetTagValue("image_tag", newTags)
+		if len(workload.Selector.Image) != 0 && len(workload.Selector.Tag) == 0 {
+			workload.Selector.Tag = "latest"
+		}
+	} else {
+		// For cgroup workloads, set service information as the selector
+		serviceName := utils.GetTagValue("service", newTags)
+		if len(serviceName) != 0 {
+			workload.Selector.Image = serviceName
+			workload.Selector.Tag = utils.GetTagValue("version", newTags)
```

Comment:
Should we expect a version tag being always present or should we set a default value here? Or Does this means if not set we'll try later on?